Update doc of admission plugin `SecurityContextDeny` to note shortcomings (#39131)
kubernetes/website: content/en/docs/reference/access-authn-authz/admission-controllers.md

### SecurityContextDeny {#securitycontextdeny}

> 

Reviewer: ```suggestion

{{< feature-state for_k8s_version="v1.0" state="alpha" >}}

```

This admission controller will deny any Pod that attempts to set the following

> [SecurityContext](/docs/reference/generated/kubernetes-api/{{< param "version">}}/#securitycontext-v1-core)

Reviewer: I would link to `/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context`

Author: Ok, why not! I should write some feedback because I think this new presentation is really worse than the old one :/

fields:
- `Pod.Spec.SecurityContext.SupplementalGroups`
- `Pod.Spec.SecurityContext.SELinuxOptions`
- `Pod.Spec.SecurityContext.RunAsUser`
- `Pod.Spec.SecurityContext.FSGroup`
- `Pod.Spec.(Init)Containers[].SecurityContext.SELinuxOptions`

> - `Pod.Spec.(Init)Containers[].SecurityContext.RunAsUser`

Reviewer: Please write these with capitalization as per the API _fields_. For example: `.spec.securityContext.supplementalGroups`.

For more historical context on this plugin, see

> [this blog post](/blog/2022/08/23/podsecuritypolicy-the-historical-context/)

Reviewer: ```suggestion
[The birth of PodSecurityPolicy ](/blog/2022/08/23/podsecuritypolicy-the-historical-context/#the-birth-of-podsecuritypolicy)
on the Kubernetes blog, which explains
```

about the PodSecurityPolicy historical context and the birth of the

> SecurityContext.

Reviewer: ```suggestion
`securityContext` field for Pods.
```

> {{< caution >}}

Reviewer: I'd put this callout just under the feature state shortcode that I also recommend adding.

Author: I agree!

This admission controller plugin is **outdated** and **incomplete**, it may be
unusable or not do what you would expect. It was originally designed to prevent
the use of some, but not all, security-sensitive fields. Indeed, fields like
`privileged`, were not filtered

> [at creation](https://github.com/kubernetes/kubernetes/pull/7343#issuecomment-97283475)

Reviewer: I wouldn't link to the issue here. The only time we like to link to issues (other than KEPs, which are a special kind of issue) is to document known issues in features that are nevertheless beta or stable - maybe also some deprecated features, at a stretch.

We rarely document lists of known issues for alpha APIs and other alpha features, because those lists are hard to maintain, often incomplete, and sometimes outright misleading.

and the plugin was not updated with the most recent fields, and new APIs like

> `EphemeralContainers`.

Reviewer: ```suggestion
the `ephemeralContainers` field for a Pod.
```
?

### SecurityContextDeny {#securitycontextdeny}

{{< feature-state for_k8s_version="v1.0" state="alpha" >}}

{{< caution >}}
This admission controller plugin is **outdated** and **incomplete**, it may be
unusable or not do what you would expect. It was originally designed to prevent
the use of some, but not all, security-sensitive fields. Indeed, fields like
`privileged`, were not filtered at creation and the plugin was not updated with
the most recent fields, and new APIs like the `ephemeralContainers` field for a
Pod.

The [Pod Security Admission](/docs/concepts/security/pod-security-admission/)
plugin enforcing the [Pod Security Standards](/docs/concepts/security/pod-security-standards/)

> restricted profile captures what this plugin was trying to achieve in a better

Reviewer 2: Nit: 
```suggestion
`Restricted` profile captures what this plugin was trying to achieve in a better
```
